obspy/io/csv/tests/test_csv.py

```diff
@@ -0,0 +1,215 @@
+import io
+import os.path
+from tempfile import gettempdir
+
+import numpy as np
+import obspy
+from obspy import read_events
+from obspy.core.util import NamedTemporaryFile
+import obspy.io.csv.core as iocsv
+from packaging import version
+import pytest
+from obspy.core.util import get_example_file
+
+
+def test_csv():
+    events = read_events()
+    with NamedTemporaryFile(suffix='.csv') as ft:
+        events.write(ft.name, 'CSV')
+        events2 = read_events(ft.name)
+    assert len(events2) == len(events)
+    assert events2[0].origins[0].time == events[0].origins[0].time
+    with NamedTemporaryFile(suffix='.csv') as ft:
+        events.write(ft.name, 'CSV', depth_in_km=True)
+        events2 = read_events(ft.name)
+    assert len(events2) == len(events)
+    assert events2[0].origins[0].time == events[0].origins[0].time
+
+
+def test_csv_reading_external_catalog():
+    names = 'year mon day hour minu sec _ lat lon dep mag id'
+    incomplete_names = 'year mon day hour minu sec _ lat lon dep'
+    fname = '/path/to/external.csv'
+    assert not iocsv._is_csv(fname)
+    events = read_events(fname, 'CSV', skipheader=1, names=names)
+    events2 = read_events(fname, 'CSV', skipheader=1, names=incomplete_names)
+    assert len(events) == 1
+    assert str(events[0].origins[0].time) == '2023-05-06T19:55:01.300000Z'
+    assert len(events[0].magnitudes) == 1
+    assert len(events2) == 1
+    assert str(events2[0].origins[0].time) == '2023-05-06T19:55:01.300000Z'
+    assert len(events2[0].magnitudes) == 0
+
+
+def test_csv_incomplete_catalog():
+    events = read_events()
+    del events[0].magnitudes[0].magnitude_type
+    events[1].magnitudes = []
+    events[1].preferred_magnitude_id = None
+    events[1].origins[0].depth = None
+    events[2].origins = []
+    events[2].preferred_origin_id = None
+    with NamedTemporaryFile(suffix='.csv') as ft:
+        with pytest.warns(Warning):
```

Comment:
Might be good to add a match argument here to make sure we are getting the right warning.

Reply:
I added a match argument. Unmatched warnings are silenced though, see https://github.com/pytest-dev/pytest/issues/9288.